avocado/plugins/spawners/remote.py

```diff
@@ -0,0 +1,210 @@
+                )
+
+        try:
+            yield slot
+        finally:
+            RemoteSpawner.slots_cache[slot] = False
+
+    @staticmethod
+    def is_task_alive(runtime_task):
+        if runtime_task.spawner_handle is None:
+            return False
+
+        session = runtime_task.spawner_handle
+
+        status, _ = session.cmd_status_output("pgrep -r R,S -f task-run")
+        return status == 0
+
+    @with_slot_reservation
+    async def spawn_task(self, runtime_task):
+        self.create_task_output_dir(runtime_task)
+        task = runtime_task.task
+        full_module_name = (
+            runtime_task.task.runnable.pick_runner_module_from_entry_point_kind(
+                runtime_task.task.runnable.kind
+            )
+        )
+        if full_module_name is None:
+            msg = f"Could not determine Python module name for runnable with kind {runtime_task.task.runnable.kind}"
+            raise RemoteSpawnerException(msg)
+        # using the "python" symlink will result in the container default python version
+        entry_point_args = ["python3", "-m", full_module_name, "task-run"]
+        entry_point_args.extend(task.get_command_args())
+
+        session = runtime_task.spawner_handle
+        LOG.info(f"Hostname: {session.host} Port: {session.port}")
+
+        setup_hook = self.config.get("spawner.remote.setup_hook")
+        # Customize and deploy test data to the container
+        if setup_hook:
+            status, output = await RemoteSpawner.run_remote_cmd_async(session, setup_hook)
+            LOG.debug(f"Customization command exited with code {status}")
+            if status != 0:
+                LOG.error(
+                    f"Error exit code {status} on {session.host}:{session.port} "
+                    f"from setup hook with output:\n{output}"
+                )
+                return False
+
+        status, output = await RemoteSpawner.run_remote_cmd_async(session, shlex.join(entry_point_args), 3600)
+        LOG.debug(f"Command exited with code {status}")
+        if status != 0:
+            LOG.error(
+                f"Error exit code {status} on {session.host}:{session.port} "
+                f"with output:\n{output}"
+                )
+            return False
+
+        return True
+
+    def create_task_output_dir(self, runtime_task):
+        output_dir_path = self.task_output_dir(runtime_task)
+        output_lxc_path = "/tmp/.avocado_task_output_dir"
+
+        os.makedirs(output_dir_path, exist_ok=True)
+        runtime_task.task.setup_output_dir(output_lxc_path)
+
+    async def wait_task(self, runtime_task):
+        while True:
+            if not RemoteSpawner.is_task_alive(runtime_task):
+                return
+            await asyncio.sleep(0.1)
+
+    async def terminate_task(self, runtime_task):
+        session = runtime_task.spawner_handle
+        status, _ = session.cmd_status_output("pkill -f task-run")
```

Comment:
It should be possible to use the task's identifier here as well.

Reply:
Ok done, resolving.